_pages/index.html

```diff
@@ -137,8 +139,12 @@ <h2>Vaše oblíbené aplikace a služby k dispozici</h2>
 					creditsWrapper.appendChild(credits);
 				}
 			}
+			titleLink.classList.remove('fade-out');
```

Comment:
Tyto změny CSS tříd by mohly jít do [`finally`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally).

Reply:
Dík, zkusím na to mrknout, ale nejsem teď u stroje. Jinak pro samotný google toho není moc potřeba, ani pro yandex, ale je tam k tomu udělaný wrapper. Možná by se ho dalo zbavit, ale sám o sobě není tak velký. Celé to vznikalo postupně, to je možná chyba. Ale samotný google-translator je celkem malý script. A ten wrapper slouží mj. k tomu, že načte jen ten překladač, který zvolíš, takže kódu je sice dost, ale nenačítá se všechno, jen to visí na serveru. A zase i nepoužívané překladače mazat mechci, kdyby google nebo i yandex vypadl. Moc jiných možností neznám. Každopádně by to určitě šlo zredukovat. Zkusím se nad tím ještě trochu zamyslet, jestli mě něco rozumného napadne. Je to celkem robustně napsané, třeba to v jedné instanci cachuje překlady, kdyby náhodou platilo, že se někdo pořád nebo vícekrát ptá na totéž, tak ať se zbytečně nevyčerpává omezený limit. Neplatí to ale pro refresh, local storage nebývá někde zpanutá, ale možná bych to mohl zkusit zaimplementovat. Tak je otázka, co s tím. Buď zcela vypustit, nebo nechat, nebo udělat ještě robustnější :), ale to by byl další kód.

Reply:
Klidně to nechcme takto a uvidíme, pokud by se limit náhodou vyčerpal.

Reply:
Ještě lépe, smazal jsem ten generic.js (asi jsem to fakt přehnal ... občas k tomu mám tendenci). Takže teď je to o kus jednodušší a jak jsem si všiml, nějaké to cachování (pouze v rámci běhu instance třídy) jsem stejně dělal na úrovni toho Google překladače, takže ten wrapper byl v podstatě zbytečný, taková další vrstva nad tím.

Z tohoto pohledu by se možná naopak další třída hodila, ale spíše ještě z druhé strany, jako podpora pro jednotlivé překladače, nikoliv mezikus mezi překladačem a stránkou. Byla by tím pásem snad i daleko jednodušší. Ale uvidíme, jestli to bude někdy potřeba. Ještě by se snad kromě local storage daly použít cookies, ale když nad tím přemýšlím, oboje asi vyžaduje na stránce nějaký GDPR notice, což ale pro ukládání překladů je snad ok.

Změna na Yandex je i tak stále jednoduchá ... na místo "google-translator.js" by mělo stačí použít "yandex-translator.js" a jinak veškerý kód zůstává. Totéž pro lokální překladač, ale kdo by to chtěl :) ... užitečné leda tak ne překlady věcí typu měsíc, den.

Jinak díky moc za ten tip s finally, rozhodně se tady hodil :).